skrf/network.py

```diff
@@ -4070,33 +4078,30 @@ def impulse_response(self, window: str = 'hamming', n: int = None, pad: int = 10
             step_response
             extrapolate_to_dc
         """
-        if n is None:
-            # Use zero-padding specification. Note that this does not allow n odd.
-            n = 2 * (self.frequency.npoints + pad - 1)
+        if bandpass is None:
+            bandpass = self.f[0] != 0
+
+        t = self.frequency._t_padded(pad=pad, n=n, bandpass=bandpass)
+        n = len(t)
 
-        fstep = self.frequency.step
-        if n % 2 == 0:
-            t = np.fft.ifftshift(np.fft.fftfreq(n, fstep))
-        else:
-            t = np.fft.ifftshift(np.fft.fftfreq(n+1, fstep))[1:]
-        if bandpass in (True, False):
-            center_to_dc = not bandpass
-        else:
-            center_to_dc = None
         if window is not None:
-            w = self.windowed(window=window, normalize=False, center_to_dc=center_to_dc)
+            w = self.windowed(window=window, normalize=False, center_to_dc=not bandpass)
         else:
             w = self
 
-        ir = mf.irfft(w.s, n=n)
+        if bandpass:
+            ir = np.fft.fftshift(np.fft.ifft(w.s, axis=0, n=n), axes=0)
```

Comment:
I was wondering why it is `ifft` and not `irfft` here, but this is coherent with https://github.com/scikit-rf/scikit-rf/pull/900#issuecomment-1526207822

In bandpass mode, there is no complex conjugate assumption like in lowpass and the center frequency of the band is just treated like the 0Hz, leading to a complex result. This `ifft` for bandpass and `irfft for lowpass`.

Reply:
Yes I remember the discussion back then, but I think the current implementation is not 100% correct, or at least my understanding back then :) 

See the last note entry here:
https://numpy.org/doc/stable/reference/generated/numpy.fft.irfft.html#numpy.fft.irfft

It's stated that the highest frequency of the `irfft` input has to be purely real, which is not the case in our application.
Usually we didn't recognise, as we use lot of padding by default. 

Metas seems to to this right, that's why I wanted to take them as reference.

Reply:
Yes, I think that the implementation you propose is correct. Also it is good to have the tests and reference data.